src/main/scala/com/codacy/rules/ReportRules.scala

```diff
@@ -38,26 +38,33 @@ class ReportRules(coverageServices: => CoverageServices) extends LogSupport {
       val filesEither = guessReportFiles(config.coverageReports, rootProjectDirIterator)
 
       filesEither.flatMap { files =>
-        val finalConfig = if (files.length > 1 && !config.partial) {
+        def sendFilesReport(partial: Boolean) = {
+          val finalConfig = config.copy(partial = partial)
+          files
+            .map { file =>
+              logger.info(s"Parsing coverage data from: ${file.getAbsolutePath} ...")
+              for {
+                _ <- validateFileAccess(file)
+                report <- CoverageParser.parse(rootProjectDir, file).map(transform(_)(finalConfig))
+                _ <- storeReport(report, file)
+                language <- guessReportLanguage(finalConfig.languageOpt, report)
+                success <- sendReport(report, language, finalConfig, commitUUID, file)
+              } yield { success }
+            }
+            .collectFirst {
+              case Left(l) => Left(l)
+            }
+            .getOrElse(Right("All coverage data uploaded."))
+        }
+        if (files.length > 1 && !config.partial) {
           logger.info("More than one file. Considering a partial report")
-          config.copy(partial = true)
-        } else config
-
-        files
-          .map { file =>
-            logger.info(s"Parsing coverage data from: ${file.getAbsolutePath} ...")
-            for {
-              _ <- validateFileAccess(file)
-              report <- CoverageParser.parse(rootProjectDir, file).map(transform(_)(finalConfig))
-              _ <- storeReport(report, file)
-              language <- guessReportLanguage(finalConfig.languageOpt, report)
-              success <- sendReport(report, language, finalConfig, commitUUID, file)
-            } yield { success }
-          }
-          .collectFirst {
-            case Left(l) => Left(l)
-          }
-          .getOrElse(Right("All coverage data uploaded."))
+          for {
+            _ <- sendFilesReport(partial = true)
+            f <- finalReport(FinalConfig(config.baseConfig))
+          } yield f
+        } else {
+          sendFilesReport(partial = true)
+        }
```

Comment:
You could simply add the if at the end to send the `finalReport`. Something like this:

```suggestion
   sendFilesReport(partial = true)
   if (files.length > 1 && !config.partial) {
       finalReport(FinalConfig(config.baseConfig))
    } else {
       sendFilesReport(partial = true)
    }
```

Reply:
The code was wrong.
In the normal case I should call `sendReport` passing the original partial setting.
And in any case you need the for comprehension because you don't want to pass the `final` if the partial send was not successful.